themes/default/content/docs/guides/pulumi-packages/schema.md

```diff
@@ -0,0 +1,321 @@
+| `dependencies`                              | `map[string]` | No       | NPM dependencies to add to package.json.                      |
+| `devDependencies`                           | `map[string]` | No       | NPM dev-dependencies to add to package.json.                  |
+| `peerDependencies`                          | `map[string]` | No       | NPM peer-dependencies to add to package.json.                 |
+| `resolutions`                               | `map[string]` | No       | NPM resolutions to add to package.json                        |
+| `typescriptVersion`                         | `string`      | No       | A specific version of TypeScript to include in package.json.  |
+| `moduleToPackage`                           | `map[string]` | No       | A map containing overrides for module names to package names. |
+| `compatibility`                             | `string`      | No       | Toggle compatibility mode for a specified target.             |
+| `disableUnionOutputTypes`                   | `boolean`     | No       | Disable support for unions in output types.                   |
+| `containsEnums`                             | `boolean`     | No       | An indicator for whether the package contains enums.          |
+
+For `python`:
+
+| Property               | Type          | Required | Description                                                                                                              |
+|------------------------|---------------|----------|--------------------------------------------------------------------------------------------------------------------------|
+| `packageName`          | `string`      | No       | PackageName is an override for the name of the generated python package.                                                 |
+| `requires`             | `map[string]` | No       | Description for the NPM package.                                                                                         |
+| `readme`               | `string`      | No       | Readme contains the text for the package's README.md files.                                                              |
+| `moduleNameOverrides`  | `map[string]` | No       | Optional overrides for Pulumi module names (e.g. `{ "flowcontrol.apiserver.k8s.io/v1alpha1": "flowcontrol/v1alpha1" }`). |
+| `compatibility`        | `string`      | No       | Toggle compatibility mode for a specified target.                                                                        |
+| `usesIOClasses`        | `boolean`     | No       | Deprecated: This bool is no longer needed since all providers now use input/output classes.                              |
+| `emitPulumiPluginFile` | `boolean`     | No       | Indicates whether the pulumiplugin.json file should be generated.                                                        |
```

Comment:
I'm tempted to not document this.

I'd originally added this option so providers could opt-in to generating the file because it requires an additional change to each provider's `Makefile` to replace the `${PLUGIN_VERSION}` string in the file with the actual version. We haven't opted-in any of our providers yet.

However, .NET and Go will soon be generating this file too (I'm prepping the PR now), and I'm planning to just go ahead and generate it by default (without an option) for those languages, and changing Python to do the same along with deleting the option.